### PR TITLE
adding hwloc-gpu variant

### PR DESCRIPTION
**Pull request by MatthewLieber**
Adds a variant that by default fixes issues where building on nodes with gpus and then trying to run on nodes with out them will fail.

Thanks to Ron Rahaman @PACE/GaTech for the report

### CODE REVIEW

**Review by natshineman — DISMISSED** [review]
This should be good to merge.

**Comment by MatthewLieber** [comment]
Can the checks please be rerun?  It looks like it might be stuck as it has been pending all weekend.

**Comment by MatthewLieber** [comment]
@alalazo All the checks have passed can we please get this merged in?

File: var/spack/repos/builtin/packages/mvapich2/package.py [hunk 1]
```diff
@@ -109,6 +109,7 @@ class Mvapich2(AutotoolsPackage):
     )
 
     variant("hwlocv2", default=False, description="Builds mvapich2 with hwloc v2")
+    variant("hwloc-graphics", default=False, description="Enables hwloc graphics")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mvapich2/package.py]: Can we do:
```suggestion
    variant("hwloc_graphics", default=False, description="Enables hwloc graphics")
```
for consistency with other variants?

File: var/spack/repos/builtin/packages/mvapich2/package.py [hunk 2]
```diff
@@ -430,7 +431,10 @@ def configure_args(self):
             args.extend(["--enable-cuda", "--with-cuda={0}".format(spec["cuda"].prefix)])
         else:
             args.append("--disable-cuda")
-
+        if "~hwloc-graphics" in self.spec:
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mvapich2/package.py]: Do we need to enable anything explicitly if this variant is active?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/mvapich2/package.py (hunk 2 above)
Comment (MatthewLieber) [reply var/spack/repos/builtin/packages/mvapich2/package.py]: Nothing needs to be done explicitly if they activate the variant.

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@MatthewLieber Can you rebase this PR? It seems stuck in CI and apparently I don't have enough permissions to rebase it myself.

**Comment by alalazo** [comment]
Closing (the PR seems to have been force pushed in a wrong way, and never fixed)

**Pull request closed without merging**